repl/src/repl.rs

```diff
@@ -362,17 +363,41 @@ fn set_globals(
             Ok(())
         }
         Pattern::Record { ref fields, .. } => {
-            let iter = fields
-                .iter()
-                .zip(::vm::dynamic::field_iter(&value, typ, vm));
-            for (field, (field_value, field_type)) in iter {
-                match field.value {
-                    Some(ref field_pattern) => {
-                        set_globals(vm, field_pattern, &field_type, &field_value)?
+            // FIXME: not sure if this is necessary, or if this has already
+            // been done. ::vm::dynamic::field_iter did this
+            // so I'm doing it too
+            let resolved_type = resolve::remove_aliases_cow(&*vm.global_env().get_env(), typ);
+
+            for pattern_field in fields.iter() {
+                let field_name: &Symbol = &pattern_field.name.value;
+                let field_value: RootedValue<&Thread> =
+                    value.get_field(field_name.declared_name())
+                    // FIXME: can I use the type checker for this error?
+                    // FIXME: apparently this error message isn't actually getting displayed. not sure why
+                    .ok_or_else(|| VMError::Message(format!(
+                        "The record doesn't have a field by the name `{}`",
+                        field_name.declared_name()
+                    )))?;
+                let field_type = match **resolved_type {
+                    Type::Record(ref row) => match **row {
+                        Type::ExtendRow {ref fields, ..} => {
```

Comment:
It is actually not necessary to match on the type here, you can instead call `resolved_type.row_iter()` and you will get a iterator over the fields (I don't know why `dynamic::field_iter` does an explicit match...)

Reply:
Thanks! Updated.